### PR TITLE
Improve install docs by mentioning how to actually download rez

### PR DESCRIPTION
**Pull request by JeanChristopheMorinPerso**
Related to https://www.bestpractices.dev/en/projects/8389#interact.

I noticed that the docs were not very clear on how rez could be download.

### CODE REVIEW

**Comment by MrLixm** [comment]
This is great !
I would also love to see, if that make sense :
- specify explicitly that python need to be installed on the system for the `python` alias to work
- the python version used with the install script determine the python version used by rez
-  which python version are preferred (more tested against I guess ?)

**Comment by parikshittiwari740** [comment]
I think the 3rd point of @MrLixm is quite important and should definitely be added :)

**Comment by JeanChristopheMorinPerso** [comment]
Thank you @MrLixm and @parikshittiwari740! I pushed a new commit that I think addresses your feedback. Does the new changes look good to you?

File: docs/source/installation.rst [hunk 1]
```diff
@@ -5,15 +5,20 @@ Installation
 Installation Script
 ===================
 
-.. warning::
-   The install script only supports Python 3.7+.
+To install rez, you will need:
+
+1. Python 3.7 or above. We support 3.7, 3.8, 3.9, 3.10 and 3.11.
+   The pythoon interpreter you use to run the install script will be the interpreter
```
Comment (brycegbrazen) [inline docs/source/installation.rst:11]: @JeanChristopheMorinPerso Small typo

**Pull request merged**